trulens_eval/trulens_eval/examples/Example_Thumbs_Application.py

```diff
@@ -0,0 +1,90 @@
+# Run with command: streamlit run Example_Application.py
+import langchain
+from langchain.callbacks import get_openai_callback
+from langchain.chains import LLMChain
+from langchain.chat_models import ChatOpenAI
+from langchain.prompts import PromptTemplate
+from langchain.prompts.chat import ChatPromptTemplate
+from langchain.prompts.chat import HumanMessagePromptTemplate
+import streamlit as st
+
+import sys
+import os
+from pathlib import Path
+dev_path = str(Path(__file__).resolve().parent.parent)
+sys.path.insert(0,dev_path)
+os.environ["OPENAI_API_KEY"] = "..."
+
+from trulens_eval import TruChain, Tru
+# Set up GPT-3 model
+model_name = "gpt-3.5-turbo"
+tru = Tru()
+# Define function to generate GPT-3 response
+@st.cache_data
+def setup_chain():
+    full_prompt = HumanMessagePromptTemplate(
+        prompt=PromptTemplate(
+            template=
+            "Provide a helpful response with relevant background information for the following: {prompt}",
+            input_variables=["prompt"],
+        )
+    )
+    chat_prompt_template = ChatPromptTemplate.from_messages([full_prompt])
+
+    chat = ChatOpenAI(model_name=model_name, temperature=0.9)
+
+    chain = LLMChain(llm=chat, prompt=chat_prompt_template)
+
+    tc = TruChain(chain, chain_id='Streamlit App')
+    tru.add_chain(chain=tc)
+    tru.run_dashboard(_dev=dev_path)
+    return tc
+
+def generate_response(prompt, tc):
+    return tc.call_with_record(prompt)
+
+tc = setup_chain()
+
+# Set up Streamlit app
+st.title("Get Help from ChatGPT")
+user_input = st.text_input("What do you need help with?")
+
+if user_input:
+    # Generate GPT-3 response
+    prompt_input = user_input
+    # add context manager to capture tokens and cost of the chain
+    with get_openai_callback() as cb:
+        gpt3_response, record = generate_response(prompt_input, tc)
+        tru.add_record(record)
```

Comment:
Why do we need to call add_record here? Isn't this going to be recorded in truchain?